txpool/journal.go

```diff
@@ -0,0 +1,91 @@
+package txpool
+
+import (
+	"sync"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+// TxStatus indicates the status of tx that is added to TxPool
+type TxStatus int8
+
+const (
+	TxSuccessful TxStatus = iota // Tx has been mined in a block successfully
+	TxDropped                    // Tx was dropped for some reason
+)
+
+var (
+	// txStatusToString is a mapping from TxStatus to text status
+	txStatusToString = map[TxStatus]string{
+		TxSuccessful: "success",
+		TxDropped:    "dropped",
+	}
+)
+
+// String returns text status from status code
+func (s *TxStatus) String() string {
+	if s == nil {
+		return ""
+	}
+
+	return txStatusToString[*s]
+}
+
+// Journal records the statuses of transactions added into TxPool
+type Journal interface {
+	// logSuccessfulTx marked the Tx as mined
+	logSuccessfulTx(txHash types.Hash)
+	// logDroppedTx marked the Tx as dropped
+	logDroppedTx(txHash types.Hash)
+	// resetTxStatus removes the status of the tx from journal
+	resetTxStatus(txHash types.Hash)
+	// txStatus returns the Tx status recorded in Journal
+	txStatus(txHash types.Hash) *TxStatus
+}
+
+// memoryJournal is an implementation of Journal and it saves data in-memory
+type memoryJournal struct {
+	txHashMap sync.Map
+}
+
+// newMemoryJournal initializes memoryJournal
+func newMemoryJournal() *memoryJournal {
+	return &memoryJournal{
+		txHashMap: sync.Map{},
+	}
+}
+
+// logSuccessfulTx marked the Tx as mined
+func (j *memoryJournal) logSuccessfulTx(txHash types.Hash) {
+	j.txHashMap.Store(txHash, true)
```

Comment:
We don't need to log successful txs.

Reply:
Do we define 'slot' the same way as in Geth?

Reply:
@JekaMas yep, https://github.com/0xPolygon/polygon-edge/blob/b97e753e8300e488e88a9046bf2aa2628bd269cc/txpool/txpool.go#L24-L25